File: packing/orderline.go
package packing


type OrderLine struct {
	Code byte
	Location int
	Cases []Case

	ndEdge *Edge		// Edge connecting the location to the I/O point
	dnEdge *Edge		// Edge connecting the I/O point to the location
}


// Constructor
func NewOrderLine (code byte, location int, cases []Case) OrderLine {
	return OrderLine{code, location, cases, nil, nil}
}

File: src/orderline.py


class OrderLine (object):
    def __init__ (self, code, location, cases=None):
        self.code = code
        self.location = location
        self.cases = cases
        
        self.pallet = None
        self.dn_edge = None
        self.nd_edge = None

    def __hash__(self):
        return hash(str(self))
